### PR TITLE
feat: get all services

### PR DESCRIPTION
**Pull request by tedim52**
## Description
```
lots_of_services = import_module(".../main.star")

def run(plan.args):
   lots_of_services.run(plan, args)
   services = plan.get_services()
   for service in services:
        <do smth>
```

## Is this change user facing?
YES

### References
https://github.com/kurtosis-tech/kurtosis/issues/2349

### CODE REVIEW

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	interpretationTimeStore *interpretation_time_value_store.InterpretationTimeValueStore
+	serviceNames            []service.ServiceName
+	description             string
+}
+
+func (builtin *GetServicesCapabilities) Interpret(_ string, arguments *builtin_argument.ArgumentValuesSet) (starlark.Value, *startosis_errors.InterpretationError) {
+	builtin.description = builtin_argument.GetDescriptionOrFallBack(arguments, descriptionStr)
+
+	// Right now, due to semantics of interpretationTimeStore this will return all services that have been added during interpretation, not accounting for services removed
+	// TODO: Remove service from interpretationTimeStore during plan.remove_service()
```
Comment (tedim52) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go]: maybe i should actually just impl this now on second thought I think it should be expected behavior

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/interpretation_time_value_store.go [hunk 2]
```diff
@@ -34,3 +34,11 @@ func (itvs *InterpretationTimeValueStore) GetService(name service.ServiceName) (
 	}
 	return serviceStarlark, nil
 }
+
+func (itvs *InterpretationTimeValueStore) GetServices() ([]*kurtosis_types.Service, error) {
+	servicesStarlark, err := itvs.serviceValues.GetServices()
+	if err != nil {
```
Comment (h4ck3rk3y) [inline core/server/api_container/server/startosis_engine/interpretation_time_value_store/interpretation_time_value_store.go:40]: lets make the error nicer!

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go [hunk 3]
```diff
@@ -62,6 +63,44 @@ func TestPutGetFail_ForMissingServiceName(t *testing.T) {
...
+	err := repository.PutService(service.ServiceName(serviceOneName), serviceOne)
+	require.Nil(t, err)
+	err = repository.PutService(service.ServiceName(serviceTwoName), serviceTwo)
+	require.Nil(t, err)
+
+	actualServices, err := repository.GetServices()
+	require.NoError(t, err)
+	require.Len(t, actualServices, 2)
+	// TODO: check the contents of each objects
+}
```
Comment (h4ck3rk3y) [inline core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go:142]: just assert that serviceOne is the same as actualServices[0]?

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go (hunk 1 above)
Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go]: i'd expect it to be running services; can we have a look at the ticket again?

how would i as the end user know what are the running services based on this list? I don't want to make exec calls to a dead service & fail

Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go]: perhaps our existing validation logic will catch the dead service being exec'd on? how do we handle

```
add A
remove A
add A
```

which A do we get?

**Review by h4ck3rk3y — APPROVED** [review]

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go (hunk 3 above)
Comment (tedim52) [reply core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go:142]: oops this wasn't meant to be left as a `TODO` def need to address

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go (hunk 1 above)
Comment (tedim52) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go]: in this scenario our existing logic `get_service` and a `get_services` call would give you get the second A, as expected.

**Pull request merged**